### ISSUE TITLE
WordSpec doesn't fully support coroutines

### ISSUE THREAD

**Issue by thibseisel** [kotest/kotest#713]
Some test scopes in `WordSpec` does not fully support calling suspending functions:

```kotlin
class WordSpecExampleTest : WordSpec() {

  init {
    "given a potato" `when` {
      delay(100) // OK
      "planted it" should {
        delay(100) // OK
        "grow" {
          delay(100) // Error: Suspension functions can be called only within coroutine body
        }
        "not be eaten".config(enabled = false) {
          delay(100) // Error: Suspension functions can be called only within coroutine body
        }
      }
    }
  }
}
```

One workaround is to wrap test code in a `launch { ... }` block (that is properly linked to the coroutine scope of `AbstractWordSpec.FinalTestContext`) but when the test fails, the assertion message is hidden behind a `JobCancelationException` which is not very readable.

**Comment by LeoColman** [comment]
Nice catch. Do you want to fix this, or leave it to us?

**Comment by thibseisel** [comment]
I'd like to, but I am unable to push my topic branch due to an HTTP 403 (Permission denied)

**Comment by LeoColman** [comment]
You need to push it to your fork, and submit a pull request to this repository from it

**Comment by sksamuel** [comment]
Fixed. Will be included in 3.3.2
